sql/moz-fx-cjms-nonprod-9a36/cjms_bigquery/flows_v1/query.sql

```diff
@@ -0,0 +1,54 @@
+WITH fxa_content_auth_stdout_events_live AS (
+  SELECT
+    PARSE_DATE('%y%m%d', _TABLE_SUFFIX) AS submission_date,
+    JSON_VALUE(jsonPayload.fields.user_properties, '$.flow_id') AS flow_id,
+    `timestamp`,
+    TO_HEX(SHA256(jsonPayload.fields.user_id)) AS fxa_uid,
+  FROM
+    `moz-fx-fxa-nonprod-375e.fxa_stage_logs.docker_fxa_auth_20*`
+  WHERE
+    jsonPayload.type = 'amplitudeEvent'
+    AND jsonPayload.fields.event_type IS NOT NULL
+    AND jsonPayload.fields.user_id IS NOT NULL
+  UNION ALL
+  SELECT
+    PARSE_DATE('%y%m%d', _TABLE_SUFFIX) AS submission_date,
+    JSON_VALUE(jsonPayload.fields.user_properties, '$.flow_id') AS flow_id,
+    `timestamp`,
+    TO_HEX(SHA256(jsonPayload.fields.user_id)) AS fxa_uid,
+  FROM
+    `moz-fx-fxa-nonprod-375e.fxa_stage_logs.docker_fxa_content_20*`
+  WHERE
+    jsonPayload.type = 'amplitudeEvent'
+    AND jsonPayload.fields.event_type IS NOT NULL
+  UNION ALL
+  SELECT
+    PARSE_DATE('%y%m%d', _TABLE_SUFFIX) AS submission_date,
+    JSON_VALUE(jsonPayload.fields.user_properties, '$.flow_id') AS flow_id,
+    `timestamp`,
+    TO_HEX(SHA256(jsonPayload.fields.user_id)) AS fxa_uid,
+  FROM
+    `moz-fx-fxa-nonprod-375e.fxa_stage_logs.stdout_20*`
+  WHERE
+    jsonPayload.type = 'amplitudeEvent'
+    AND jsonPayload.fields.event_type IS NOT NULL
+)
+SELECT
+  submission_date,
+  flow_id,
+  MIN(`timestamp`) AS flow_started,
+  ARRAY_AGG(
+    IF(fxa_uid IS NULL, NULL, STRUCT(fxa_uid, `timestamp` AS fxa_uid_timestamp)) IGNORE NULLS
+    ORDER BY
+      `timestamp` DESC
+    LIMIT
+      1
+  )[SAFE_OFFSET(0)].*,
```

Comment:
I think only taking the last FxA per flow might miss some edge cases.

For example, if someone subscribes to VPN, but then in the same session logs into a different FxA after that for some reason (e.g. shared computer) this would currently only get the FxA that didn't subscribe.

Reply:
I will pass that on, but this is the compromise I worked out with subplat eng for selecting a single fxa_uid per flow.